lfs/pointer.go

```diff
@@ -195,7 +195,7 @@ func parseOid(value string) (string, error) {
 	if parts[0] != oidType {
 		return "", errors.New("Invalid Oid type: " + parts[0])
 	}
-	oid := parts[1]
+	oid := strings.TrimSpace(parts[1])
```

Comment:
It's unclear to me why we need this.  Can you say a few words about this change?

Reply:
@bk2204 I was concerned some LFS pointer files might have trailing space, newlines, or line feed characters (e.g. with Windows) now that we have a `\z` in the regexp. I think `decodeKVData` calls `strings.SplitN`, which might keep `\r`?

Reply:
Ah, that makes sense.  I believe `bufio.NewScanner` strips off all of the possible line ending characters, so that shouldn't be a problem and we can just remove this line.  Pointer files should not have CRLF line feeds, since they're supposed to be bit-for-bit unique, but I believe we'll handle it gracefully even if they do.

Reply:
@bk2204 Now I remember why I added this. Do we care about tolerating trailing spaces? This now fails:

```diff
diff --git a/lfs/pointer_test.go b/lfs/pointer_test.go
index 7960bd38..88ab2e58 100644
--- a/lfs/pointer_test.go
+++ b/lfs/pointer_test.go
@@ -88,8 +88,7 @@ func TestDecodeTinyFile(t *testing.T) {
 
 func TestDecode(t *testing.T) {
 	ex := `version https://git-lfs.github.com/spec/v1
-oid sha256:4d7a214614ab2935c943f9e0ff69d22eadbb8f32b1258daaa5e2ca24d17e2393
-size 12345`
+oid sha256:4d7a214614ab2935c943f9e0ff69d22eadbb8f32b1258daaa5e2ca24d17e2393` + "  \nsize 12345"
 
 	p, err := DecodePointer(bytes.NewBufferString(ex))
 	assertEqualWithExample(t, ex, nil, err)
```

Reply:
No, we definitely do not want people to use trailing spaces.  We want people to use the single, canonical pointer file without any modifications.  The fact that we happen to tolerate CRs in the file is an oversight, not a feature.